BaseClasses.py

```diff
@@ -336,7 +336,10 @@ def get_placeable_locations(self, state=None, player=None) -> List[Location]:
 
     def get_unfilled_locations_for_players(self, location_name: str, players: Iterable[int]):
         for player in players:
-            location = self.get_location(location_name, player)
+            if (location_name, player) in self._location_cache:
```

Comment:
why is this change made?

Reply:
IIRC get_location results in an error if the location doesn't exist for that player, so it is checking to see if the location exists first